packages/shorten-url/src/DidCommShortenUrlService.ts

```diff
@@ -0,0 +1,84 @@
+import { CredoError, EventEmitter, InboundMessageContext, injectable } from '@credo-ts/core'
+
+import {
+  InvalidateShortenedUrlReceivedEvent,
+  RequestShortenedUrlReceivedEvent,
+  ShortenUrlEventTypes,
+  ShortenedUrlReceivedEvent,
+} from './DidCommShortenUrlEvents'
+import { RequestShortenedUrlMessage, ShortenedUrlMessage, InvalidateShortenedUrlMessage } from './messages'
+
+@injectable()
+export class DidCommShortenUrlService {
+  public constructor(private readonly eventEmitter: EventEmitter) {}
+
+  // Create messages to send
+  public createRequest(options: {
+    url: string
+    goalCode: string
+    requestedValiditySeconds: number
+    shortUrlSlug?: string
+  }) {
+    if (!Number.isInteger(options.requestedValiditySeconds) || options.requestedValiditySeconds < 0) {
+      throw new CredoError('requested_validity_seconds MUST be a non-negative integer')
+    }
+    return new RequestShortenedUrlMessage(options)
+  }
+
+  public createShortenedUrl(options: { threadId: string; shortenedUrl: string; expiresTime?: number }) {
+    return new ShortenedUrlMessage(options)
+  }
+
+  public createInvalidate(options: { shortenedUrl: string; threadId?: string }) {
+    return new InvalidateShortenedUrlMessage(options)
+  }
+
+  // Process inbound messages and emit events
+  public async processRequest(ctx: InboundMessageContext<RequestShortenedUrlMessage>) {
```

Comment:
```suggestion
  public async processRequest(inboundMessageContext: InboundMessageContext<RequestShortenedUrlMessage>) {
```

To follow usual patterns we have in credo-ts